clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java

```diff
@@ -138,6 +147,9 @@ public ByteBuffer allocate(int size, long maxTimeToBlockMs) throws InterruptedEx
                             recordWaitTime(timeNs);
                         }
 
+                        if (this.closed)
```

Comment:
In the while loop, lock is held.

How would the close() call wake up the await() on line 143 ?